feat: filter results via IN statements with multiple values

retorted opened the pull request:
Proposal / draft: adds a new method `where_in` for filtering on a column by a given set of values. Eg:
```rust

let selected_flavours = vec!["chocolate", "vanilla", "mint"];
let ice_creams = IceCream::all().where_in(|ic| ic.flavour, selected_flavours);

=> SELECT * FROM ice_creams WHERE flavour IN ("chocolate","vanilla","mint")
```

What do you think?

lex148 commented:
@retorted, This is great! yeah we need a where_in style query. I do have a a couple thoughts here.

- I would prefer to not give ownership when calling to create the `IN` clause. Being able to support passing a slice would be so nice and more convenient for the user. 
- I think I like this inside the clauses better. Specifically because that is where all the other operators are too. "<", ">" ,  "ILIKE", ...

I moved your merge request to point to a different branch so I can show you what I am thinking.

retorted commented:
> Being able to support passing a slice would be so nice
:+1:

> I think I like this inside the clauses better.

Ha, my first iteration put it in the clauses, but I changed my mind and switched it into the `QueryBuilder` before pushing.
